src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisState.cs

```diff
@@ -234,6 +234,7 @@ private void OnCompilationEventsGenerated_NoLock(ImmutableArray<CompilationEvent
             AddToEventsMap_NoLock(compilationEvents, filterTreeOpt);
 
             // Mark the events for analysis for each analyzer.
+            List<ISymbol> newPartialSymbols = null;
```

Comment:
Usually we use pooled ArrayBuilders. Is there a specific reason to use List<T> here?